### DISCUSSION TITLE
What to do when a content dialog is bigger than parent window?

### DISCUSSION THREAD

**Discussion by Prochy** [microsoft/WindowsAppSDK#4309] [Q&A]
Is there any possible solution how to solve problem when a window, which shows a content dialog, is smaller than the content dialog itself? If this happens the ContentDialog is cropped. 
It came to me two possible solutions:
1. Temporary increase size of the parent window
2. Create a custom window for ContentDialog and don't use embedded ContentDialog.

But maybe there is better way how to solve that I'm not seeing.

![image](https://github.com/microsoft/WindowsAppSDK/assets/6740126/c4b1f43c-a2c2-4edf-9579-a60655ba76e7)
![image](https://github.com/microsoft/WindowsAppSDK/assets/6740126/c08d5b49-9f92-4c58-8e0b-47af5e5e996c)

**Comment by Yewzir** [comment]
@Prochy 

The first thing that came to mind was using a [popup](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.primitives.popup) dialog instead. However, I just looked a few things up in the documentation, and I think, if you want to stick to a content dialog, the `ContentDialog.FullSizeDesired` property is just what you need.

> ### [ContentDialog.FullSizeDesired Property](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.contentdialog.fullsizedesired)
> 
> **Definition**
> 
> Gets or sets a value that indicates whether a request is being made to display the dialog full size.
> 
> **Remarks**
> 
> In a desktop app, a dialog is sized to its content by default. Requesting that it be shown at full size changes the behavior to size the dialog based on set height and width proportions. The dialog does not cover the whole screen.

Since its default value is `false`, assigning `true` should render your dialog as per your requirement.

🎉 

Kind regards
Yewzir

**Reply by Prochy** [reply]
@Yewzir thanks for trying helping me but unfortunately none of those help to solve my issue. 
I need to resize the base window, the FullSizeDesired property only affects the content dialog size but the base window remains same after opening the dialog.

![image](https://github.com/user-attachments/assets/5b745f94-57a5-411f-bb83-82b666747a91)

And I want a base window to get resized when opening dialog like this: 
![image](https://github.com/user-attachments/assets/0e14ae36-828f-43c8-9b84-b8576c8371a6)

**Reply by Yewzir** [reply]
@Prochy 

I'm afraid I'm not understanding your issue all too well in that case. If you want the parent window to have greater dimensions than the content dialog, then it's just a matter of specifying those greater dimensions on the parent window right from the beginning, no? At least, that is how I am interpreting it.

Edit: Oh, you want the parent to scale with the child when the child is shown, and then return the parent to its original dimensions. Did I get that right?

> And I want a base window to get resized when opening dialog like this:
> ![349426468-0e14ae36-828f-43c8-9b84-b8576c8371a6](https://github.com/user-attachments/assets/a5aa2406-e277-4d43-af76-5c6a66cc0584)

**Reply by Prochy** [reply]
Yes, exactly, but I already solved, I do it programmatically in backed and after closing dialog, the window size get back.
